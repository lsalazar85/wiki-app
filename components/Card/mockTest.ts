import { URL_LOADER_IMG } from "../../constants/urls";

export const mockTestCharacter = {
  id: "1",
  name: "Nick",
  species: "Human",
  status: "Alive",
  episode: ["1", "2", "3"],
  image: URL_LOADER_IMG,
  gender: "Male",
};

export const mockTestEpisode = {
  id: "1",
  name: "Nick",
  species: "Human",
  air_date: "December 10",
  episode: "Episode 1",
  characters: ["Rick"],
};
