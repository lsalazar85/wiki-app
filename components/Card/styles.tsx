import styled from 'styled-components';
import Image from 'next/image';
import colors from '../../styles/colors';

export const CardWrapper = styled.div`
  display: flex;
  border-radius: 0.5rem;
  border: 1px solid ${colors.Orange};
  max-height: 15rem;

  @media (min-width: 320px) {
    width: 100%;
    margin: 0 0 1rem 0;
  }

  @media (min-width: 1000px) {
    flex: 1 1 48%;

    &:nth-child(even) {
      margin-left: 1rem;
    }
  }
`;

export const CardImage = styled(Image)`
  border-top-left-radius: 0.5rem;
  border-bottom-left-radius: 0.5rem;
`;

export const CardInfo = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 0.5rem;
`;

export const Name = styled.h2``;

export const ItemInfo = styled.span<{ cursorStyle?: boolean }>`
  font-size: 0.95rem;
  cursor: ${(props) => (props.cursorStyle ? 'pointer' : 'auto')};
  color: ${(props) =>
    props.cursorStyle ? `${colors.Orange}` : `${colors.Black}`};

  &:last-child {
    margin-top: 1rem;
  }
`;

export const DetailsList = styled.div`
  font-size: 0.9rem;
  padding: 0.5rem;
`;

export const DetailText = styled.p``
