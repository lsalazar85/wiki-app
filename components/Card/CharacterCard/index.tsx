import { useState } from 'react';
import { CardWrapper, CardImage, CardInfo, Name, ItemInfo } from '../styles';

import Modal from '../../Modal';
import ModalEpisodesList from '../EpisodeCard/ModalEpisodesList';

import { CharacterCardProps } from '../../../interfaces';

const CharacterCard = ({ data }: CharacterCardProps) => {
  const [modalState, setModalState] = useState<boolean>(false);

  return (
    <CardWrapper>
      <CardImage src={data.image} alt={data.name} width={210} height={210} />
      <CardInfo>
        <Name>{data.name}</Name>
        <ItemInfo>Status: {data.status}</ItemInfo>
        <ItemInfo>Gender: {data.gender}</ItemInfo>
        <ItemInfo>Specie: {data.species}</ItemInfo>
        <ItemInfo cursorStyle onClick={() => setModalState((prev) => !prev)}>
          Character Episodes
        </ItemInfo>
      </CardInfo>
      <Modal
        visible={modalState}
        title={`${data.name} Episodes`}
        onClose={() => setModalState(false)}
      >
        {data?.episode.map((url) => {
          const urlParts = url.split('/').filter(Boolean);
          const episodeId = urlParts[urlParts.length - 1];

          return (
            <ModalEpisodesList key={`episode-${episodeId}`} id={episodeId} />
          );
        })}
      </Modal>
    </CardWrapper>
  );
};

export default CharacterCard;
