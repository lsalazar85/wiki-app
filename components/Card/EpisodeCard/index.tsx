import { useState } from 'react';
import { CardWrapper, CardInfo, Name, ItemInfo } from '../styles';

import Modal from '../../Modal';
import ModalCharactersList from '../../Modal/ModalCharactersList';

import { EpisodeCardProps } from '../../../interfaces';

const EpisodeCard = ({ data }: EpisodeCardProps) => {
  const [modalState, setModalState] = useState<boolean>(false);

  return (
    <CardWrapper>
      <CardInfo>
        <Name>{data.name}</Name>
        <ItemInfo>Name: {data.name}</ItemInfo>
        <ItemInfo>Air Date: {data.air_date}</ItemInfo>
        <ItemInfo>Episode: {data.episode}</ItemInfo>
        <ItemInfo cursorStyle onClick={() => setModalState((prev) => !prev)}>
          Characters in Episodes
        </ItemInfo>
      </CardInfo>
      <Modal
        visible={modalState}
        title={`Ep. ${data.name} Characters`}
        onClose={() => setModalState(false)}
      >
        {data?.characters.map((url) => {
          const urlParts = url.split('/').filter(Boolean);
          const characterId = urlParts[urlParts.length - 1];

          return (
            <ModalCharactersList
              key={`character-${characterId}`}
              id={characterId}
            />
          );
        })}
      </Modal>
    </CardWrapper>
  );
};

export default EpisodeCard;
