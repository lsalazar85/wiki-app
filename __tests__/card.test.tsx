/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react'
import Card from "../components/Card";


describe('Card', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<Card data={}/>)

        expect(asFragment()).toMatchSnapshot();
        expect(asFragment()).toBeDefined()
    })
})
