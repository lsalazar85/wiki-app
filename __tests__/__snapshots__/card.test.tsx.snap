// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render successfully 1`] = `
<DocumentFragment>
  <div
    class="styles__CardContainer-sc-tyfa84-0 ominj"
  >
    <div
      class="styles__ImageContainer-sc-tyfa84-1 grvmgA"
    >
      <div
        style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="display: block; box-sizing: border-box; padding-top: 100%;"
        />
        <img
          alt="Rick Sanchez"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
